vyper/parser/external_call.py

```diff
@@ -76,9 +77,13 @@ def external_call(node, context, interface_name, contract_address, pos, value=No
                 ],
             ]
         )
+    if output_size:
+        # when return data is expected, revert if the length is insufficient
+        static_output_size = get_static_size_of_type(sig.output_type) * 32
+        sub.append(["assert", ["gt", "returndatasize", static_output_size - 1]])
```

Comment:
What if it returns bytes?

Probably needs separate cases if it's static or dynamic... Perhaps done in another PR